### PR TITLE
Clamp vassalize amount to the amount of people in the department, closes #13785

### LINKED ISSUE #13785: Bloodsuckers can get objectives they can't possibly complete.
### Severity

Round Breaking

### Short Description

Bloodsuckers have the ability to get an objective that is not possible for them to complete.  In this case, Kaitlyn Wilsow (ckey: AJK737) got the objective to have 3 vassals in the security department, despite there only being one person even existing in sec roundstart (and throughout the rest of the round as well) 

### Reproduction

1. Start round, roll bloodsucker
2. Lowpop shift, with nobody but you, a couple assistants, and one engineer.
3. Get the objective 'have four vassals in engineering'
4. Fail to complete this objective due to obvious reasons

here's an image of the crew manifest and Kaitlyn's objectives:
[https://i.postimg.cc/yNVnFfpr/image-2022-04-19-020642592.png](url)

### Round ID

Round #40590

### Test Merges

#13419, #13435, #13321

### BYOND Username

Cark

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Simplehorror**
# Document the changes in your pull request
Fixes #13785
# Changelog
:cl:  
bugfix: Bloodsuckers don't get impossible objectives
/:cl:

### CODE REVIEW

**Comment by Simplehorror** [comment]
Not sure how to test this locally but should work in theory

File: code/modules/antagonists/bloodsuckers/bloodsucker_objectives.dm [hunk 1]
```diff
@@ -93,15 +93,24 @@
...
 	switch(rand(0, 2))
 		// Vasssalize Command/QM
 		if(0)
 			target_amount = 1
 			target_department = VASSALIZE_COMMAND
 		// Vassalize a certain department
 		else
-			target_amount = rand(2,3)
 			target_department = pick(departments)
+			target_amount = clamp(rand(2,3), 0, LAZYLEN(departments_to_list[target_department]))
```
Comment (Bibby0110) [inline code/modules/antagonists/bloodsuckers/bloodsucker_objectives.dm:113]: departments_to_list only contains references to lists of what jobs there are, not how many people are actually playing in that department

Comment (Simplehorror) [reply code/modules/antagonists/bloodsuckers/bloodsucker_objectives.dm:113]: going to harvest the check completion code

**Comment by ghost** [comment]
> Not sure how to test this locally but should work in theory

Open multiple windows of the game.

**Pull request closed without merging**